assets/prototype/application/ui/components/display/espressoIcon/EspressoIcon.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Dashicon } from '@wordpress/components';
+import { Path, SVG } from '@wordpress/components/src/primitives';
+
+import ICONS from './constants';
+import PATHS from './svgPaths';
+import EspressoIconProps from './types';
+
+/**
+ * Custom Event Espresso Dashicons
+ */
+const EspressoIcon: React.FunctionComponent<EspressoIconProps> = ({ icon, svgSize = 20, className, ...otherProps }) => {
+	let path: string;
+	switch (icon) {
+		case ICONS.CALCULATOR:
+			path = PATHS.CALCULATOR;
+			break;
+
+		case ICONS.CALENDAR:
+			path = PATHS.CALENDAR;
+			break;
+
+		case ICONS.REM:
+			path = PATHS.REM;
+			break;
+
+		case ICONS.ROTATE:
+			path = PATHS.ROTATE;
+			break;
+
+		case ICONS.SAVE:
+			path = PATHS.SAVE;
+			break;
```

Comment:
It looks like a great candidate for an enum.

Reply:
oh yeah I guess that would have been a better idea
i used enums for the button too, but didn't think to do so here!